src/baldr/incident_singleton.h

```diff
@@ -149,20 +159,22 @@ struct incident_singleton_t {
    * The thread begins by deciding whether its just scaning the directory (works for a small number of
    * incidents) or using a memory mapped changelog to communicate about which incidents chnaged last.
    * If the latter is used we preallocate
-   * @param config
-   * @param tileset
-   * @param state
+   * @param config     lets the function know where to look for incidents and desired update frequency
+   * @param tileset    if the tileset is static we can use lockfree mode
+   * @param state      used for interthread communication
+   * @param interrupt  functor that, if set and returns true, stops the main loop of this function
```

Comment:
in order to unit test this i needed a way to control when the function exists. in practice we never want the function to exit but for testing its quite annoying to have an infinite loop. so we allow an interrupt function here to stop the loop when desired